api/authorization-api-1_0.md

```diff
@@ -888,165 +889,97 @@ The following is a non-normative example of a response to an Access Evaluations
 
-## Subject Search Semantics
+To perform a search, the client provides the Subject, Resource, Action, and Context entities defined in the Information Model ({{information-model}}), but omits the unique identifier of the entity being queried. The authorization service (PDP) then responds with the set of authorized entities for the queried entity type which would be authorized according to the provided criteria.
 
-While the evaluation of a search is implementation-specific, it is expected that any returned results that are then fed into an `evaluation` call MUST result in a `"decision": true` response.
+## Semantics {#search-semantics}
 
-In addition, it is RECOMMENDED that a subject search is performed transitively, traversing intermediate attributes and/or relationships. For example, if the members of group G are designated as viewers on a document D, then a search for all users that are viewers of document D will include all the members of group G.
+A search is designed to return entities that would correspond to a permitted decision. Therefore, any result from a Search API, when subsequently used in an Access Evaluation API call, SHOULD result in a `"decision": true` response. However, because the evaluation is implementation-specific and may depend on other variables (such as time), this outcome is not guaranteed.
 
-## The Subject Search API Request {#subject-search-request}
+In addition, it is RECOMMENDED that a search be performed transitively, traversing intermediate attributes and/or relationships. For example, if user U is a member of group G, and group G is designated as a viewer on a document D, then a search for all subjects that can view document D will include user U.
 
-The Subject Search request is an object consisting of the following entities:
+## Pagination {#search-pagination}
 
-`subject`:
-: REQUIRED. The subject (or principal) of type Subject.  NOTE that the Subject type is REQUIRED but the Subject ID can be omitted, and if present, is IGNORED.
+Search APIs can return large result sets. To manage this, a Policy Decision Point (PDP) MAY support pagination, allowing clients to navigate and retrieve subsets of the total result set. 
 
-`action`:
-: REQUIRED. The action (or verb) of type Action.
+Pagination does not guarantee an atomic snapshot of the result set. Consequently, if items are added or removed while paginating, results MAY be repeated or omitted between pages.
 
-`resource`:
-: REQUIRED. The resource of type Resource.
+Pagination is based on the use of opaque tokens. A client makes an initial request for data by sending a query that does not contain a token. In response, the PDP MAY initiate pagination by returning a partial result set instead of the complete one.
```

Comment:
Pagination is based on the use of opaque tokens. A client makes an initial request for data by sending a query that does not contain a token. If the PDP determines that the response contains too many results to fit in a single response, the PDP returns a partial result set and a token that the caller can use to retrieve the next page of results.